Source/SwiftLintFramework/Rules/EmptyParenthesesWithTrailingClosureRule.swift

```diff
@@ -0,0 +1,84 @@
+//
+//  EmptyParenthesesWithTrailingClosureRule.swift
+//  SwiftLint
+//
+//  Created by Marcelo Fabri on 11/12/16.
+//  Copyright © 2016 Realm. All rights reserved.
+//
+
+import Foundation
+import SourceKittenFramework
+
+public struct EmptyParenthesesWithTrailingClosureRule: ASTRule, ConfigurationProviderRule {
+    public var configuration = SeverityConfiguration(.warning)
+
+    public init() {}
+
+    public static let description = RuleDescription(
+        identifier: "empty_parentheses_with_trailing_closure",
+        name: "Empty Parentheses with Trailing Closure",
+        description: "When using trailing closures, empty parenthesis should be avoided " +
+                     "after the method call.",
+        nonTriggeringExamples: [
+            "[1, 2].map { $0 + 1 }\n",
+            "[1, 2].map({ $0 + 1 })\n",
+            "[1, 2].reduce(0) { $0 + $1 }",
+            "[1, 2].map { number in\n number + 1 \n}\n",
+            "let isEmpty = [1, 2].map.isEmpty()\n"
+        ],
+        triggeringExamples: [
+            "[1, 2].map↓() { $0 + 1 }",
+            "[1, 2].map↓( ) { $0 + 1 }\n",
+            "[1, 2].map↓() { number in\n number + 1 \n}\n",
+            "[1, 2].map↓(  ) { number in\n number + 1 \n}\n"
+        ]
+    )
+
+    public enum Kind: String {
```

Comment:
this is the second place where this enum is needed and the 3rd where `"source.lang.swift.expr.call"` is used. Should we create a shared enum? If so, where?